presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/MergeLimitWithDistinct.java

```diff
@@ -53,6 +53,7 @@
                         child.getSource(),
                         parent.getCount(),
                         false,
+                        child.getSource().getOutputSymbols(),
```

Comment:
This should be the aggregation node's group-by keys. Otherwise, it would transform this:

```
- Limit ::= [a,b,c]
    - Aggregation[a,b,c] ::= [a, b, c]
        - Project ::= [a,b,c,d]
```

into

```
- DistinctLimit[a,b,c,d] ::= [a,b,c,d]
    - Project ::= [a,b,c,d]
```